crates/nu-command/src/conversions/into/filesize.rs

```diff
@@ -116,20 +116,18 @@ pub fn action(input: &Value, _args: &CellPathOnlyArgs, span: Span) -> Value {
                 val: 0,
                 span: value_span,
             },
-            _ => Value::Error {
-                error: ShellError::UnsupportedInput(
-                    "'into filesize' for unsupported type".into(),
+            other => Value::Error {
+                error: ShellError::OnlySupportsThisInputType(
+                    "string and integer".into(),
+                    other.get_type().to_string(),
+                    span,
                     value_span,
                 ),
             },
         }
     } else {
-        Value::Error {
-            error: ShellError::UnsupportedInput(
-                "'into filesize' for unsupported type".into(),
-                span,
-            ),
-        }
+        // Propagate existing errors
+        input.clone()
```

Comment:
I'm a little iffy on this assumption that no span will always mean an error - feels like the kind of thing that might cause trouble in the future. Not a big enough deal to be a blocker though.